astropy/units/format/generic.py

```diff
@@ -464,15 +473,72 @@ def _parse_unit(cls, s, detailed_exception=True):
         else:
             raise ValueError()
 
+    _translations = {
+        '\N{GREEK SMALL LETTER MU}': '\N{MICRO SIGN}',
+        '\N{MINUS SIGN}': '-',
+        '\N{GREEK CAPITAL LETTER OMEGA}': '\N{OHM SIGN}',
+        '\N{LATIN CAPITAL LETTER A WITH RING ABOVE}': '\N{ANGSTROM SIGN}'
+    }
+    # key in the translations dict must be the codepoint
+    _translations = {ord(k):v for k, v in _translations.items()}
+    """Character translations that should be applied before parsing a string.
+
+    Note that this does explicitly *not* generally translate MICRO SIGN to u,
+    since then a string like 'µ' would be interpreted as unit mass.
+    """
+
+    _superscripts = (
+        '\N{SUPERSCRIPT MINUS}'
+        '\N{SUPERSCRIPT PLUS SIGN}'
+        '\N{SUPERSCRIPT ZERO}'
+        '\N{SUPERSCRIPT ONE}'
+        '\N{SUPERSCRIPT TWO}'
+        '\N{SUPERSCRIPT THREE}'
+        '\N{SUPERSCRIPT FOUR}'
+        '\N{SUPERSCRIPT FIVE}'
+        '\N{SUPERSCRIPT SIX}'
+        '\N{SUPERSCRIPT SEVEN}'
+        '\N{SUPERSCRIPT EIGHT}'
+        '\N{SUPERSCRIPT NINE}'
+    )
+
+    _superscript_translations = str.maketrans(_superscripts, '-+0123456789')
+    _regex_superscript = re.compile(f'[{_superscripts}]+')
+    _regex_deg = re.compile('°([CF])?')
+
+    @classmethod
+    def _convert_superscript(cls, m):
+        return '({})'.format(
+            m.group().translate(cls._superscript_translations)
+        )
+
+    @classmethod
+    def _convert_deg(cls, m):
+        if len(m.string) == 1:
+            return 'deg'
+        return m.group().replace('°', 'deg_')
+
     @classmethod
     def parse(cls, s, debug=False):
         if not isinstance(s, str):
             s = s.decode('ascii')
+        elif not s.isascii():
+            # Translate some basic unicode items that we'd like to support on
+            # input but are not standard.
+            s = s.translate(cls._translations)
+
+            # TODO: might the below be better done in the parser/lexer?
+            # Translate superscripts to parenthesized numbers; this ensures
+            # that mixes of superscripts and regular numbers fail.
+            s = cls._regex_superscript.sub(cls._convert_superscript, s)
+            # Translate possible degrees.
+            s = cls._regex_deg.sub(cls._convert_deg, s)
 
         result = cls._do_parse(s, debug=debug)
         # Check for excess solidi, but exclude fractional exponents (accepted)
-        if (s.count('/') > 1 and
-                s.count('/') - len(re.findall(r'\(\d+/\d+\)', s)) > 1):
+        n_slashes = s.count('/')
+        n_fractions = len(re.findall(r'\(\d+/\d+\)', s))
+        if n_slashes > 1 and (n_slashes - n_fractions) > 1:
```

Comment:
Why this change? You now always look for fractions, which will slow down the parsing for all units. I'd at least keep it inside an `if` statement.